src/ConfigurationClientManager.ts

```diff
@@ -0,0 +1,296 @@
+const MINIMAL_CLIENT_REFRESH_INTERVAL = 30 * 1000; // 30 seconds in milliseconds
+const SRV_QUERY_TIMEOUT = 30 * 1000; // 30 seconds in milliseconds
+
+export class ConfigurationClientManager {
+    isFailoverable: boolean;
+    dns: any;
+    endpoint: string;
+    #secret : string;
+    #id : string;
+    #credential: TokenCredential;
+    #clientOptions: AppConfigurationClientOptions | undefined;
+    #appConfigOptions: AzureAppConfigurationOptions | undefined;
+    #validDomain: string;
+    #staticClients: ConfigurationClientWrapper[];
+    #dynamicClients: ConfigurationClientWrapper[];
+    #lastFallbackClientRefreshTime: number = 0;
+    #lastFallbackClientRefreshAttempt: number = 0;
+
+    constructor (
+        connectionStringOrEndpoint?: string | URL,
+        credentialOrOptions?: TokenCredential | AzureAppConfigurationOptions,
+        appConfigOptions?: AzureAppConfigurationOptions
+    ) {
+        let staticClient: AppConfigurationClient;
+
+        if (typeof connectionStringOrEndpoint === "string" && !instanceOfTokenCredential(credentialOrOptions)) {
+            const connectionString = connectionStringOrEndpoint;
+            this.#appConfigOptions = credentialOrOptions as AzureAppConfigurationOptions;
+            this.#clientOptions = getClientOptions(this.#appConfigOptions);
+            staticClient = new AppConfigurationClient(connectionString, this.#clientOptions);
+            const ConnectionStringRegex = /Endpoint=(.*);Id=(.*);Secret=(.*)/;
+            const regexMatch = connectionString.match(ConnectionStringRegex);
+            if (regexMatch) {
+                this.endpoint = regexMatch[1];
+                this.#id = regexMatch[2];
+                this.#secret = regexMatch[3];
+            } else {
+                throw new Error(`Invalid connection string. Valid connection strings should match the regex '${ConnectionStringRegex.source}'.`);
+            }
+        } else if ((connectionStringOrEndpoint instanceof URL || typeof connectionStringOrEndpoint === "string") && instanceOfTokenCredential(credentialOrOptions)) {
+            let endpoint = connectionStringOrEndpoint;
+            // ensure string is a valid URL.
+            if (typeof endpoint === "string") {
+                try {
+                    endpoint = new URL(endpoint);
+                } catch (error) {
+                    if (error.code === "ERR_INVALID_URL") {
+                        throw new Error("Invalid endpoint URL.", { cause: error });
+                    } else {
+                        throw error;
+                    }
+                }
+            }
+
+            const credential = credentialOrOptions as TokenCredential;
+            this.#appConfigOptions = appConfigOptions as AzureAppConfigurationOptions;
+            this.#clientOptions = getClientOptions(this.#appConfigOptions);
+            staticClient = new AppConfigurationClient(connectionStringOrEndpoint.toString(), credential, this.#clientOptions);
```

Comment:
Move it to the bottom of this else-if branch.
And 
```suggestion
            staticClient = new AppConfigurationClient(this.endpoint.toString(), this.#credential, this.#clientOptions);
```